src/server.h

```diff
@@ -2027,6 +2027,8 @@ struct valkeyServer {
     unsigned long long cluster_link_msg_queue_limit_bytes;  /* Memory usage limit on individual link msg queue */
     int cluster_drop_packet_filter; /* Debug config that allows tactically
                                    * dropping packets of a specific type */
+    int cluster_close_link_on_packet_drop; /* Debug config that goes along with cluster_drop_packet_filter.
```

Comment:
I would really suggest we start naming debug flags in a distinct way, either prefix or suffix works for me. This helps readability. Not a blocker of this PR though since we have many existing debug flags already.

Reply:
Additionally, it'd be enough to use one bit instead of 64. How about `int debug_flags` for all these?

Reply:
I had a different opinion which is that I think these debug flags should be entirely compiled out of our production build. They can use as much memory as they want, that also makes it clearer they are debug related, since they won't be available.

Reply:
I'm not sure I agree. It's a good habit to run make test before make install. For make test, you need the debug functionality.

Reply:
> It's a good habit to run make test before make install. 

I have never done this. If it is a thing people do, I guess I retract my position.

Reply:
I don't know if people do it. :)

Anyway, I think there's some value to be able to run tests (ours or others) on an unmodified build.

Reply:
I have added a prefix of `debug_` and using 1 bit for the new flag. The flags are currently sandwiched between two 8 byte types, so the flag will anyway get 8 bytes. 

I didn't touch `int cluster_drop_packet_filter` as it is initialized with -1 to avoid dropping PING packets. It can probably be set to `CLUSTERMSG_TYPE_COUNT`, but that can change when we add a new message type requiring a code change. It cannot be one bit.

```c
#define CLUSTERMSG_TYPE_PING 0          /* Ping */
#define CLUSTERMSG_TYPE_PONG 1          /* Pong (reply to Ping) */
#define CLUSTERMSG_TYPE_MEET 2          /* Meet "let's join" message */
#define CLUSTERMSG_TYPE_FAIL 3          /* Mark node xxx as failing */
#define CLUSTERMSG_TYPE_PUBLISH 4       /* Pub/Sub Publish propagation */
#define CLUSTERMSG_TYPE_FAILOVER_AUTH_REQUEST 5 /* May I failover? */
#define CLUSTERMSG_TYPE_FAILOVER_AUTH_ACK 6     /* Yes, you have my vote */
#define CLUSTERMSG_TYPE_UPDATE 7        /* Another node slots configuration */
#define CLUSTERMSG_TYPE_MFSTART 8       /* Pause clients for manual failover */
#define CLUSTERMSG_TYPE_MODULE 9        /* Module cluster API message. */
#define CLUSTERMSG_TYPE_PUBLISHSHARD 10 /* Pub/Sub Publish shard propagation */
#define CLUSTERMSG_TYPE_COUNT 11        /* Total number of message types. */
```

Reply:
## When using `uint32_t`

```
/*   5328      |       4 */    int cluster_module_flags;
/*   5332      |       4 */    int cluster_allow_reads_when_down;
/*   5336      |       4 */    int cluster_config_file_lock_fd;
/* XXX  4-byte hole      */
/*   5344      |       8 */    unsigned long long cluster_link_msg_queue_limit_bytes;
/*   5352      |       4 */    int cluster_drop_packet_filter;
/*   5356      |       4 */    uint32_t debug_cluster_close_link_on_packet_drop;
/*   5360      |       8 */    mstime_t busy_reply_threshold;
/*   5368      |       4 */    int pre_command_oom_state;
/*   5372      |       4 */    int script_disable_deny_script;
```

## When using 1 bit with `uint32_t`

```
/*   5328      |       4 */    int cluster_module_flags;
/*   5332      |       4 */    int cluster_allow_reads_when_down;
/*   5336      |       4 */    int cluster_config_file_lock_fd;
/* XXX  4-byte hole      */
/*   5344      |       8 */    unsigned long long cluster_link_msg_queue_limit_bytes;
/*   5352      |       4 */    int cluster_drop_packet_filter;
/*   5356: 0   |       4 */    uint32_t debug_cluster_close_link_on_packet_drop : 1;
/* XXX  7-bit hole       */ 
/* XXX  3-byte hole      */
/*   5360      |       8 */    mstime_t busy_reply_threshold;
/*   5368      |       4 */    int pre_command_oom_state;
/*   5372      |       4 */    int script_disable_deny_script;
```

## When using `uint16_t`

```
/*   5328      |       4 */    int cluster_module_flags;
/*   5332      |       4 */    int cluster_allow_reads_when_down;
/*   5336      |       4 */    int cluster_config_file_lock_fd;
/* XXX  4-byte hole      */
/*   5344      |       8 */    unsigned long long cluster_link_msg_queue_limit_bytes;
/*   5352      |       4 */    int cluster_drop_packet_filter;
/*   5356      |       2 */    uint16_t debug_cluster_close_link_on_packet_drop;
/* XXX  2-byte hole      */
/*   5360      |       8 */    mstime_t busy_reply_threshold;
/*   5368      |       4 */    int pre_command_oom_state;
/*   5372      |       4 */    int script_disable_deny_script;
```

## When using 1 bit with `uint16_t`

```
/*   5328      |       4 */    int cluster_module_flags;
/*   5332      |       4 */    int cluster_allow_reads_when_down;
/*   5336      |       4 */    int cluster_config_file_lock_fd;
/* XXX  4-byte hole      */
/*   5344      |       8 */    unsigned long long cluster_link_msg_queue_limit_bytes;
/*   5352      |       4 */    int cluster_drop_packet_filter;
/*   5356: 0   |       2 */    uint16_t debug_cluster_close_link_on_packet_drop : 1;
/* XXX  7-bit hole       */
/* XXX  3-byte hole      */
/*   5360      |       8 */    mstime_t busy_reply_threshold;
/*   5368      |       4 */    int pre_command_oom_state;
/*   5372      |       4 */    int script_disable_deny_script;
```

Reply:
There are many ints in the server struct that are essentially booleans. I think we can allow ourselves to rearrange the fields within the same section. We could do this:

```C
    int cluster_allow_replica_migration:1; /* Automatic replica migrations to orphaned masters and from empty masters */
    int cluster_require_full_coverage:1;   /* If true, put the cluster down if
                                              there is at least an uncovered slot.*/
    int cluster_slave_no_failover:1;       /* Prevent slave from starting a failover
                                              if the master is in failure state. */
    int cluster_allow_reads_when_down:1;   /* Are reads allowed when the cluster
                                              is down? */
```

There are probably many bytes to save, but let's not dive too deep in this, since it's a singleton.

Reply:
Agree. Will keep this new flag as single bit and leave further optimizations out of this PR. Thanks!